zerver/context_processors.py

```diff
@@ -35,6 +35,18 @@ def common_context(user: UserProfile) -> Dict[str, Any]:
         'user_name': user.full_name,
     }
 
+def common_context_unauthed(realm: Realm) -> Dict[str, Any]:
+    """Common context used for things like outgoing emails that don't
+    have a request and which are not yet authenticated.
+    """
+    return {
+        'realm_uri': realm.uri,
+        'realm_name': realm.name,
+        'root_domain_uri': settings.ROOT_DOMAIN_URI,
+        'external_uri_scheme': settings.EXTERNAL_URI_SCHEME,
+        'external_host': settings.EXTERNAL_HOST
+    }
```

Comment:
This should be extracted from `common_context` - have `common_context` use this new helper for getting these values - so that we avoid copypasted code :slightly_smiling_face:

Reply:
I got a bit confused. So what I understand is that we have to do is something like this:-
```
def common_context(user: UserProfile) -> Dict[str, Any]:
    return {
        'context': common_context_unauthed(user.realm),
        'user_name': user.full_name,
    }
```
and `common_context_unauthed` remains the same as it is above. Am I right or am I missing something?

Reply:
Yes, though the correct syntax will be something more like
```
context = common_context_unauthed(user.realm)
context.update({'user_name': ...})
```